content/hardware/04.pro/boards/portenta-h7/tutorials/secure-boot/secure-boot.md

```diff
@@ -0,0 +1,71 @@
+---
+title: 'Secure Boot on Portenta H7'
+description: 'Learn how to use secure boot on the Arduino Portenta H7.'
+tags: 
+  - Secure Boot
+author: 'Umberto Baldi'
+hardware:
+  - hardware/04.pro/boards/portenta-h7
+
+software:
+  - ide-v1
+  - ide-v2
+  - cli
+---
+## Introduction
+This short tutorial will guide the user through enabling the secure boot on the Portenta H7, generating custom security keys, and using them with the MCUboot bootloader.
+
+## Hardware & Software Required
+-   [Portenta H7](https://store.arduino.cc/portenta-h7)
+-   Arduino IDE 1.8.19+  or Arduino IDE 2.0.0-rc5+ (https://www.arduino.cc/en/software)
+-   [Arduino Mbed OS Portenta Boards](https://github.com/arduino/ArduinoCore-mbed) version 3.0.0+
+-   [imgtool](https://github.com/arduino/imgtool-packing/releases/latest) (optional)
+
+## Instructions
+
+### Flashing the Latest Bootloader
+In order to have secure boot enabled you must update the bootloader on your Portenta H7 and use [MCUboot](https://www.mcuboot.com/). You can find more info on how to perform the update in [this other tutorial](https://docs.arduino.cc/tutorials/portenta-h7/updating-the-bootloader).
+
+### Use Default Security Keys
+Once The bootloader has been updated to MCUboot, it's possible to use [secure boot](https://www.keyfactor.com/blog/what-is-secure-boot-its-where-iot-security-starts/) to have an additional layer of security. From that point on it is required to upload a compiled sketch with the Custom Board Option **"Security settings"** set to **"Signature + Encryption"** (the option can be found under **Tools > Security settings** in the IDE when selecting Portenta H7 as board, or you can use `--board-options security=sien` if using the Arduino CLI). Failing to provide such option will cause the bootloader not to run the compiled sketch because it is not trusted.
+
+If no operation is performed the default security keys are used.
+These keys are embedded in the example sketch `STM32H747_updateBootloader` which can be found in  **Files > Examples > STM32H747_System > STM32H747_updateBootloader**. A private 256bit [ECDSA](https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm) key is used for the encryption (`ecdsa-p256-encrypt-key.h`), while a public key is used for the signing (`ecdsa-p256-signing-key.h`). These two keys are the ones the bootloader uses to verify if a sketch is valid or not, before starting it for the first time. The default private keys used after compiling a sketch to sign and encrypt it are located in `Arduino15/packages/arduino/hardware/mbed_portenta/<version>/libraries/MCUboot/default_keys/`.
```

Comment:
How can a public key be used for signing? Shouldn't a private key be used for signing and the public one for verifying the signature?

Reply:
https://github.com/arduino/ArduinoCore-mbed/pull/447#issue-1195653112 CC @pennam

Reply:
since we are referring to the keys included in the bootloader maybe we can rewrite the paragraph like this:

Two keys are embedded in the example sketch `STM32H747_updateBootloader` which can be found in  **Files > Examples > STM32H747_System > STM32H747_updateBootloader** and used by the bootloader.
A private 256bit [ECDSA](https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm) key is used to extract the encryption key and decrypt the binary update (`ecdsa-p256-encrypt-key.h`),  while the public key is used for image verification (`ecdsa-p256-signing-key.h`).

As counterpart, when building the image update, imgtool uses this private [key](https://github.com/arduino/ArduinoCore-mbed/pull/447/files#diff-f43e4850d60c61854678f6f80c6ddc4b59e3e68ca7e71b02e5ed15288c9aadb4) to sign the image and this public [key](https://github.com/arduino/ArduinoCore-mbed/pull/447/files#diff-95bb7b27de14276896a2bec099dc5a498d5332616458c04263efc8d24810e6a6) for image encryption with elliptic curve integrated encryption scheme.